api/base/parsers.py

```diff
@@ -25,14 +50,31 @@ def parse(self, stream, media_type=None, parser_context=None):
         data = result.get('data', {})
 
         if data:
-            if 'attributes' not in data:
-                raise JSONAPIException(source={'pointer': '/data/attributes'}, detail=NO_ATTRIBUTES_ERROR)
+            relationships = data.get('relationships')
+
+            path = stream.path
+            related_resources = ['contributors', 'node_links']
```

Comment:
I second this question. Hardcoding related resource assumptions in the generic top level parser (`JSONAPIParser`) seems like an invitation to trouble.

If we need special related resources, we can subclass the parser, add relationship info to the serializer class, or something else- but the generic parser should not implement highly view-specific logic.